include/nlohmann/detail/iterators/iteration_proxy.hpp

```diff
@@ -40,15 +44,23 @@ template<typename IteratorType> class iteration_proxy_value
     /// a string representation of the array index
     mutable string_type array_index_str = "0";
     /// an empty string (to return a reference for primitive values)
-    const string_type empty_str{};
+    string_type empty_str{};
```

Comment:
Why is this no longer const?

Reply:
To get around the issue of not being able to default the copy assignment operator. I thought that was the simplest solution of all. `static` isn't one unless we disable the exit-time destructor warnings.

Reply:
Ah, right.  That's annoying.  It means that we don't get the compiler protection against something accidentally modifying it.

> static isn't one unless we disable the exit-time destructor warnings.

What is this referring to?

Reply:
> > static isn't one unless we disable the exit-time destructor warnings.
> 
> What is this referring to?

It's referring to me having had a bad time using statics in the abandoned "Improve unit testing (2)" PR. Long story short, statics of non-trivial type involve destructor calls on program exit, that can cause issues so it's good practice to not use them and also turn on warning flags for them (and they are for this project). There are workarounds but i'd rather stay clear of them.

Reply:
Okay.  I wonder if just writing out the copy assignment operator by hand is still the best solution.

Reply:
Yes, it can make sure that we don't accidentally write code that modifies it at some point in the future.  I think that there are probably optimizations too, but that's less of a concern.